packages/gasket-plugin-dynamic-plugins/lib/index.js

```diff
@@ -9,8 +9,8 @@ export default {
   name,
   version,
   hooks: {
-    async prepare(gasket) {
-      if (!gasket.config.dynamicPlugins?.length) return;
+    async prepare(gasket, config) {
+      if (!gasket.config.dynamicPlugins?.length) return config;
```

Comment:
Since we are making `prepare` a waterfall lifecycle, you should probably modify and return the incoming `config` object here instead of mutating `gasket.config` as it will end up getting stomped over.

```suggestion
      if (!config.dynamicPlugins?.length) return config;
```

Basically, replace `gasket.config` -> `config` throughout this hook